components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java

```diff
@@ -358,6 +358,10 @@ protected void startPolling() {
             if (LOG.isTraceEnabled()) {
                 LOG.trace("The kafka consumer was woken up while polling on thread {} for {}", threadId, getPrintableTopic());
             }
+        } catch (Error e) { // NOSONAR - rethrown
```

Comment:
I kinda forgot whether the order of the catch matters, eg I would assume this should be last, so regular exception can be caught for the normal exceptions

Reply:
Yeah, if we were to implement this ... it would have to be the last one, indeed.

Reply:
@orpiske Although it would not resolve the problem entirely, capturing the error and implementing appropriate logging would provide valuable insights to the developer regarding the occurrences of poll timeouts and the consumer leaving the group.

Reply:
@orpiske We found the root cause during our usage of the Spring Kafka framework (KafkaMessageListenerContainer as the listener for message consumption). However, thanks to the error handling and logging capabilities provided by the framework, we were able to capture and log the errors, despite not yet determining the root cause.

Reply:
The problem is: just because it worked on your particular scenario, there's no guarantee it would work reliably (or work at all) on all others that fall on the same category (i.e.: all other sub-classes of Error).